components/steering_wheel/include/IO/IO_Digital.h

```diff
@@ -0,0 +1,58 @@
+/**
+ * IO_Digital.h
+ * Digital IO Module Header
+ * @note To transfer this module to another system, the defined FLAGS and the
+ *      defined NUM_INPUTS must be modified to be congruent with the new system
+ */
+
+#pragma once
+
+/******************************************************************************
+ *                             I N C L U D E S
+ ******************************************************************************/
+
+#include "stm32f103xb.h" /**< Needed for GPIO_TypeDef */
+#include "Types.h" 
+
+
+/******************************************************************************
+ *                              D E F I N E S
+ ******************************************************************************/
+
+#define SW1_FLAG            0x01
+#define SW2_FLAG            0x02
+#define SW4_FLAG            0x04
+#define SW5_FLAG            0x08
+#define BTN_TOP_LEFT_FLAG   0x10
+#define BTN_MIDDLE_FLAG     0x20
+#define BTN_TOP_RIGHT_FLAG  0x40
+#define NUM_INPUTS          7U /**< Must be equal to the count of flags */
+
+/******************************************************************************
+ *                             T Y P E D E F S
+ ******************************************************************************/
+
+typedef 
+#if 8 >= NUM_INPUTS
```

Comment:
Why? What's the point of this? What is the benefit of this vs. just specifically choosing the type and adding a static assert to ensure that the pins will fit in that type?

Reply:
More scalable/reusable from what I think, to transfer this module to another system would only take changing the defined flags and NUM_INPUTS of the IO_Digital.h file. Less to adapt long term

Reply:
but how is that more portable than just making a variable of the correct size by hand?

Reply:
It would calculate automatically based on enum size so could capitalize on that note. Will edit to directly setting it manually